lib/two_factor_authentication/models/two_factor_authenticatable.rb

```diff
@@ -30,14 +50,15 @@ def authenticate_otp(code, options = {})
           totp.verify_with_drift(code, drift)
         end
 
-        def otp_code(time = Time.now, options = {})
+        def generate_totp_code(time = Time.now, options = {})
+          raise "generate_totp_code called with no otp_secret_key set" if otp_secret_key.nil?
```

Comment:
I don't think this is necessary. This method would only be called in tests, so it's up to the tests to make sure the user is totp_enabled. If anything, I would remove this method from this module and just make it available as a test helper because it should never be called in the app.